client/layout/masterbar/item.tsx

```diff
@@ -221,49 +216,30 @@ class MasterbarItem extends Component< MasterbarItemProps > {
 			disabled: this.props.disabled,
 		};
 
-		if ( this.props.url && ! this.props.subItems ) {
-			return (
+		const MenuItem = ( props: React.HTMLAttributes< HTMLElement > ) =>
```

Comment:
FYI it's not a good idea to create components inside a render method. It causes the component to be recreated on every render which can have side effects ([like this](https://github.com/Automattic/wp-calypso/issues/98893)). See https://github.com/Automattic/wp-calypso/pull/98897 which should fix this issue by moving the component's definition outside the render method.